Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/OneTimePassword.kt

```diff
@@ -0,0 +1,18 @@
+package de.rki.coronawarnapp.datadonation
+
+import androidx.annotation.Keep
+import com.google.gson.annotations.SerializedName
+import org.joda.time.Instant
+import java.util.UUID
+
+@Keep
+data class OneTimePassword(
+    @SerializedName("uuid")
+    val uuid: UUID = UUID.randomUUID(),
+    @SerializedName("time")
+    val time: Instant = Instant.now()
```

Comment:
Don't we also have to store the `expirationDate` (that we get after the `otp` authentication) and the time when the `otp` authentication was performed?

See point 1 and 2 in the tech-spec [here](https://github.com/corona-warn-app/cwa-app-tech-spec/blob/068f948fdadfda84712e6f23f79d792e6103e946/docs/spec/data-donation-client-edus.md#authorization-request-of-otp)

Reply:
I would rather dynamically calculate the expiration date and not set it at creation time.

Reply:
to me the success and time of auth would be part of a request entity, not the otp itself

Reply:
Thanks!

Reply:
> I would rather dynamically calculate the expiration date and not set it at creation time.

We don't calculate the `expirationDate`, we get it from the backend on a successful authentication, or?

> to me the success and time of auth would be part of a request entity, not the otp itself

But we have to persist those values, right?

Reply:
you're right! :)